<img src="../figs/logoIC.png" width="585" alt="image_0.png">

# Cálculo (G1953) - Grado en Ingeniería Civil

#### Guión original en MatLab de Vera Egorova <vera.egorova@unican.es>, adaptado para Octave por Joaquín Bedia <bediaj@unican.es>
#### Dpto. Matemática Aplicada y Ciencias de la Computación

<!--
<a name="T_55FB4C11"></a>
-->

***

# <span style="color:rgb(213,80,0)">**Introducción a OCTAVE**</span>

<p style="text-align:left">
   <img src="https://upload.wikimedia.org/wikipedia/commons/thumb/6/6a/Gnu-octave-logo.svg/240px-Gnu-octave-logo.svg.png" width="75">
</p>

***


## Contenido

<a name="beginToc"></a>

&emsp;[¿Qué es OCTAVE?](#H_11636A56)

&emsp;[Empezando a usar OCTAVE](#H_134FF586)

&emsp;[Variables](#H_D10F85D2)

&emsp;&emsp;[Tipos de datos](#H_230C9F48)

&emsp;[Comandos básicos](#H_B1D664E6)

&emsp;&emsp;[Cabecera de un programa](#H_C3E5E6CE)

&emsp;&emsp;[Operaciones básicas](#H_F4823953)

&emsp;&emsp;[Funciones matemáticas elementales](#H_7133A156)

&emsp;&emsp;[Formatos](#H_A5416571)

&emsp;[Matrices y vectores](#H_0C8E9485)

&emsp;&emsp;[Crear y manipular matrices](#H_AD12C960)

&emsp;&emsp;[Matrices especiales](#H_E94A2E5C)

&emsp;&emsp;[Submatrices](#H_A56524CE)

&emsp;&emsp;[Operaciones con matrices](#H_A84CB336)

&emsp;&emsp;[Comandos sum y prod](#H_935550B0)

&emsp;[Ejercicios resueltos](#H_703E17D1)

&emsp;[Ejercicios propuestos](#H_A998F915)

&emsp;[Resumen](#H_0B4541F7)

&emsp;[Soluciones ](#H_AC6B0A09)

<a name="endToc"></a>
<a name="H_0CC5F680"></a>
<a name="H_11636A56"></a>

***


### ¿Qué es OCTAVE?

[Octave](https://octave.org) es un lenguaje de programación específicamente diseñado para cálculo numérico con vectores y matrices. Constituye una inferfaz de alto nivel para un buen número de librerías estándar de cálculo numérico como LAPACK o BLAS.

La sintaxis de Octave es similar a la de MatLab. En la gran mayoría de los casos (y probablemente en todas las prácticas de cálculo numérico que realizaréis durante el Grado), los programas escritos en Octave pueden ejecutarse sin ninguna modificación en MatLab. Dado que MatLab es un software comercial, éste posee un mayor número de funciones que Octave, así como paquetes especializados (*tool-boxes*) no disponibles en Octave, y en algunas ocasiones de indiscutible potencia.

No obstante, la funcionalidad requerida para la mayoría de las prácticas del grado está disponible igualmente en Octave. Como contrapartida, Octave es [*software libre*](https://www.gnu.org/philosophy/free-sw.es.html), lo que quiere decir no sólo que es gratuíto, sino que, a diferencia de MatLab, otorga al usuario las llamadas 4 libertades sobre el programa que utiliza:

 * Libertad de usar el programa, con cualquier propósito.
 * Libertad de acceso al código fuente del programa.
 * Libertad de copiado y distribución.
 * Libertad de modificar y mejorar el programa.

Existen gran cantidad de manuales de ayuda que permitirán al alumno progresar tanto como desee en sus habilidades como programador en Octave/MatLab. Como primera aproximación, se sugiere el [manual introductorio](http://math.jacobs-university.de/oliver/teaching/iub/resources/octave/octave-intro/octave-intro.html) de Hubert Selhofer *et al.*. Además, los alumnos cuentan con un [manual de Octave/Matlab](http://meteo.unican.es/work/juaco/G328_intro_Octave.pdf) adaptado a los requerimientos iniciales para esta asignatura.
Por último, mencionar una breve guía para comenzar a usar OCTAVE incluída en la propia wiki del proyecto: [Using Octave](https://wiki.octave.org/Using_Octave).


<a name="H_134FF586"></a>

### Empezando a usar OCTAVE

La ventana de OCTAVE instalada en nuestro equipo presenta una interfaz gráfica que muestra un escritorio dividido en varias partes, entre ellas:

-  Las órdenes a ejecutar se escriben en la ventana de comandos, <samp>Command Window</samp>.
-  En el editor (ventana <samp>Editor</samp>) escribimos código que guardaremos en scripts con extensión <samp>.m</samp> y ejecutaremos en la ventana de comandos.
-  La ventana <samp>Workspace</samp> proporciona información sobre las variables utilizadas.

... si bien esta disposición es configurable por parte del usuario, por lo que puede presentarse en muy diversas formas

<p style="text-align:left">
   <img src="../figs/octave_GUI_ES.png" width="750">
</p>


Aunque hay más, los tipos de datos básicos en OCTAVE para nosotros son los siguientes: números reales o complejos, booleanos (valores lógicos) y cadenas de caracteres (o <samp>strings</samp>). En cualquier caso, el objeto básico de trabajo en OCTAVE son las matrices. Los escalares y los vectores son considerados casos particulares de matrices.


**Uso de la ventana de comandos**


Las instrucciones que terminan por punto y coma no imprimen los resultados en la pantalla. Se pueden recuperar instrucciones anteriores, usando las teclas de flechas arriba y abajo, o bien acudiendo a la ventana con el historial de la sesión (también es posible guardar el historial y reabrirlo posteriormente, aunque generalmente preferimos utilizar un script y guardarlo antes de cerrar para poder volver a ejecutar nuestro programa en otra ocasión.).

<a name="H_D10F85D2"></a>

### Variables

Para crear una nueva variable con la que trabajar, se especifica el nombre que se le quiere dar a dicha variable, seguido del signo igual " <samp>="</samp> (operador de asignación) y el valor que se le desea asignar.


Hay ciertas **reglas para definir variables en OCTAVE**:

-  No se pueden crear variables sin asignarles un valor.
-  El nombre de una variable puede tener como máximo 63 caracteres que pueden ser letras, números y el guion de subrayar.
-  El primer carácter tiene que ser una letra, `matriz2` es un nombre válido, pero no lo es `2matriz`.
-  Las mayúsculas y las minúsculas tienen valor distintivo. La variable `Modulo` es distinta de la variable `modulo`.
-  Dentro de un nombre de variable no puede haber espacios en blanco, `modulo1`  es un nombre de variable válido, pero `modulo 1` NO.
-  Existen nombres que deben evitarse porque tienen significado propio en OCTAVE (son _variables reservadas_): `ans`, `pi`, `Inf`, `i`, por ejemplo... 

Introduce en la línea de comandos la siguiente instrucción, y presiona la tecla `Enter` para ejecutar el código:

In [38]:
Pi = pi

Pi = 3.1416


Aparece el resultado: hemos creado una variable <samp>Pi</samp> que tiene el valor 3.1416. También la variable aparece en el espacio de trabajo (Workspace):

<p style="text-align:left">
   <img src="../figs/workspace_Pi.png" width="436">
</p>


A tener en cuenta:

-  Al abrir sesión seleccionaremos el **directorio de trabajo**, donde almacenaremos los scripts.
-  Si se escribe <samp>;</samp> al final de una linea de código, <u>no se  muestra el resultado por pantalla</u>. Esta característica es la deseada cuando se trabaja con grandes conjuntos de datos:

In [39]:
a = 1; % Se crea una variable a con valor 1, pero no se imprime el resultado
b = 2  % Se imprime el resultado (b = 2)

b = 2


-  **Obtener ayuda** en OCTAVE: El comando <samp>help</samp> es la orden  básica de ayuda con la cual consultar el uso de una  función, incluyendo los argumentos de entrada/salida, una breve  descripción de qué realiza y de su uso.

In [40]:
help pi

'pi' is a built-in function from the file libinterp/corefcn/data.cc

 -- pi
 -- pi (N)
 -- pi (N, M)
 -- pi (N, M, K, ...)
 -- pi (..., CLASS)
     Return a scalar, matrix, or N-dimensional array whose elements are
     all equal to the ratio of the circumference of a circle to its
     diameter.

     Internally, 'pi' is computed as '4.0 * atan (1.0)'.

     When called with no arguments, return a scalar with the value of
     pi.

     When called with a single argument, return a square matrix with the
     dimension specified.

     When called with more than one scalar argument the first two
     arguments are taken as the number of rows and columns and any
     further arguments specify additional matrix dimensions.

     The optional argument CLASS specifies the return type and may be
     either "double" or "single".

     See also: e, I.

Additional help for built-in functions and operators is
available in the online version of the manual.  Use the command
'doc <topic>' to sear


-  Se pueden incluir **comentarios** (aclaraciones que no se ejecutan) dentro de los programas añadiendo <samp>%</samp> delante de lo que se desea comentar.

In [41]:
% ESTO ES UN COMENTARIO. A continuación muestro por pantalla el valor de pi:
pi

ans = 3.1416



-  Se puede **ejecutar** un programa entero pulsando la tecla "Run", o bien una(s) línea(s) seleccionadas pulsando F9 o seleccionando "Evaluate selection" en el menú que sale al pinchar con el botón derecho del ratón.
-  Es **esencial leer los errores** que obtenemos cuando algo no funciona, ya que contienen información acerca del error. Además, la(s) línea(s) afectada(s) tiene una señal roja a la derecha del editor.

In [42]:
a1 = 1; % Ok
% Descomenta la siguiente línea (Elimina el símbolo de comentario %) y observa qué sucede:
% 1a = 1 % PARSE ERROR! Octave da la explicación (y a veces ofrece una solución). Observa el símbolo '^' que indica el punto en el que ha sucedido el error


**Ejercicio:** Descomenta las siguentes líneas del código y corrige los errores:


In [43]:
% 1a = 1
% valor 1 = pi
% x = % 5

<a name="H_230C9F48"></a>

#### Tipos de datos

OCTAVE trabaja con los tipos básicos de datos (números, caracteres y  booleanos) a partir de los cuales se construyen vectores, matrices,  cadenas de caracteres, etc. Del mismo modo, a partir del tipo de datos se definen diferentes operaciones básicas (suma, producto, intersecciones, etc.) que permiten la construcción y definición por composición de operaciones complejas.


Además de los tipos de datos básicos existen diferentes tipos de  variables que sirven de contenedores de datos, como son el tipo <samp>cell</samp>, <samp>struct</samp> o variables simbólicas (<samp>sym</samp>).


Para saber qué variables están definidas y a que tipo de datos corresponde tenemos el comando <samp>whos:</samp>


In [44]:
whos

Variables visible from the current scope:

variables in scope: top scope

  Attr   Name               Size                     Bytes  Class
  ====   ====               ====                     =====  ===== 
         A                  2x3                         48  double
         B                  2x3                         48  double
         C                  2x3                         48  double
         M                  3x3                         72  double
         Pi                 1x1                          8  double
         a                  1x1                          8  double
         a1                 1x1                          8  double
         ans                1x1                          8  double
         b                  1x1                          8  double
         b1                 1x1                          8  double
         b2                 1x1                          8  double
         c                  1x1                         


Compara con el contenido de la ventana <samp>Workspace</samp>. Como mencionamos anteriormente, los escalares son un caso particular de matrices; tienen un tamaño de 1x1, lo que significa que son matrices con una fila y una columna.

<a name="H_B1D664E6"></a>

### Comandos básicos
<a name="H_C3E5E6CE"></a>

***
#### Cabecera de un programa
***

In [45]:
% Limpia el espacio de trabajo
clear all % se recomienda usar 'clearvars' para borrar las variables
% Limpia la ventana de comandos
clc
% Cierra las ventanas/figuras abiertas
close all
% Lista las variables definidas en el espacio de trabajo
whos

<a name="H_F4823953"></a>

***
#### Operaciones básicas
***

La forma de representar números y de operar con OCTAVE es la misma que  la de una calculadora de bolsillo, usando las operaciones \texttt{+, \-, \*, /, ^}. Se recomienda **siempre asignar el resultado a una variable**. Si no asignas el resultado a una variable, OCTAVE lo almacena temporalmente en la variable predeterminada <samp>ans</samp>, pero esto puede llevar a errores y confusión, especialmente en scripts más complejos.


In [46]:
suma = 17+5 % Suma
division = 17/5 % Division con decimales

suma = 22
division = 3.4000


<a name="H_7133A156"></a>

***
#### Funciones matemáticas elementales
***

-  <samp>exp(x)</samp>: exponencial de x
-  <samp>log(x)</samp>: logaritmo natural o neperiano.
-  <samp>log10(x)</samp>: logaritmo en base 10.
-  <samp>sin(x)</samp>: seno de x.
-  <samp>cos(x)</samp>: coseno de x.
-  <samp>tan(x)</samp>: tangente de x.
-  <samp>asin(x)</samp>: arco seno de x con imagen en el rango $[-\pi /2,\pi /2]$ .
-  <samp>acos(x)</samp>: arco coseno de x con imagen en el rango $[-\pi /2,\pi /2]$ .
-  <samp>atan(x)</samp>: arco tangente de x con imagen en el rango $[-\pi /2,\pi /2]$ .
-  <samp>atan2(y,x)</samp>: arco tangente de y/x con imagen en el rango $[-\pi ,\pi ]$ .
-  <samp>sinh(x)</samp>: seno hiperbólico de x.
-  <samp>cosh(x)</samp>: coseno hiperbólico de x.
-  <samp>tanh(x)</samp>: tangente hiperbólica de x.
-  <samp>sum(x)</samp>: suma de los elementos del vector x.
-  <samp>prod(x)</samp>: producto de los elementos del vector x.

 **Importante:** en estas funciones el ángulo viene en radianes:


In [47]:
c = cos(pi) % no 'cos(180)' para calcular el coseno de pi (de 180 grados)
a = 4*atan(1) % atan: Funcion arcotangente con resultado en radianes.
s = sin(pi/2) % Seno de pi/2
l = log(exp(1)^5) % Función logaritmo natural

c = -1
a = 3.1416
s = 1
l = 5



Jerarquia de las operaciones: multiplicacion y division, suma y resta:


In [48]:
b1 = 1+2*3
b2 = (1+2)*3 % Uso del parentesis

b1 = 7
b2 = 9


<a name="H_A5416571"></a>

***
#### Formatos
***

Podemos cambiar la forma en la que OCTAVE muestra por pantalla los números mediante el comando <samp>format</samp>:


In [49]:
format short, 1/3  % 4 decimales
format long, 1/3  % 15 decimales
format longe, 1/3  % Notación científica larga
format rat, 1/3  % fraccion

ans = 0.3333
ans = 0.333333333333333
ans = 3.333333333333333e-01
ans = 1/3



Sin embargo, hay que tener en cuenta que el cambio de formato sólo tiene efecto a nivel de visualización (el número con el que OCTAVE trabaja en memoria seguirá siendo siempre el mismo).


In [50]:
format short

<a name="H_0C8E9485"></a>
### **Matrices y vectores**


<a name="H_DDAE71AC"></a>
***
#### Crear y manipular vectores
***

Hay varias maneras de construir vectores fila:


In [51]:
u=[-7,3,4,-2] % Corchetes y comas
v=[-7 3 4 -2] % Corchetes sin comas
t= 1:3:20 % para crear secuencias de números tomados de tres en tres,
% tales que inicial:incremento:final
z= linspace(1,2,9) % genera un vector fila de 9 elementos cuyo primer elemento
% es 1 y el último 2, siendo todos sus elementos equidistantes:
% linspace(inicial, final, n)

u =

  -7   3   4  -2

v =

  -7   3   4  -2

t =

    1    4    7   10   13   16   19

z =

 Columns 1 through 8:

   1.0000   1.1250   1.2500   1.3750   1.5000   1.6250   1.7500   1.8750

 Column 9:

   2.0000




Y vectores columna:


In [52]:
w=[-7;3;4;-2] % Vector vertical (uso de ';' como fin de línea)
x= u' % Trasponiendo un vector fila

w =

  -7
   3
   4
  -2

x =

  -7
   3
   4
  -2




Se puede comprobar el tamaño de los vectores y extraer elementos:


In [53]:
lu = length(u) % Numero de elementos del vector
lw = length(w)
su = size(u) % Numero de filas y columnas

u(1) % Primer elemento del vector u
w(4) % Cuarto elemento del vector w
u(1)=-4 % Cambia el primer elemento del vector u, y lo reemplaza por el valor -4

lu = 4
lw = 4
su =

   1   4

ans = -7
ans = -2
u =

  -4   3   4  -2



<a name="H_AD12C960"></a>

***
#### Crear y manipular matrices
***

In [54]:
A=[3,-7;2,5] %Corchetes y comas. Definida por filas
B=[3 -7; 2 5] %Corchetes sin comas
C=[6 0 -3 8 9; 5 1 7 -4 2; 3 1 -6 2 -1]
C(2, 3) %Elemento de la fila 2 y columna 3 de la matriz c
size(C) %Numero de filas y columnas de la matriz c
length(C) %Numero de columnas de la matriz c
C(1, 4)=11 %Cambio del elemento de la fila 1 y 4 columna de la matriz c

A =

   3  -7
   2   5

B =

   3  -7
   2   5

C =

   6   0  -3   8   9
   5   1   7  -4   2
   3   1  -6   2  -1

ans = 7
ans =

   3   5

ans = 5
C =

    6    0   -3   11    9
    5    1    7   -4    2
    3    1   -6    2   -1



<a name="H_E94A2E5C"></a>

***
#### Matrices especiales
***

Son aquellas formadas por ceros, unos o la identidad.


In [55]:
zeros(3,5) %Nula de 3 filas y 5 columnas
zeros(4) %Nula 4x4
ones(2,4) %Unidad de 2 filas y 4 columnas
ones(3) %Unidad 3x3
eye(4) % Matriz identidad de orden 4 (i.e. diagonal de 1 4x4)
A = diag([-7 2 4]) %Diagonal con los valores dados

ans =

   0   0   0   0   0
   0   0   0   0   0
   0   0   0   0   0

ans =

   0   0   0   0
   0   0   0   0
   0   0   0   0
   0   0   0   0

ans =

   1   1   1   1
   1   1   1   1

ans =

   1   1   1
   1   1   1
   1   1   1

ans =

Diagonal Matrix

   1   0   0   0
   0   1   0   0
   0   0   1   0
   0   0   0   1

A =

Diagonal Matrix

  -7   0   0
   0   2   0
   0   0   4



<a name="H_A56524CE"></a>

***
#### Submatrices
***

Se pueden extraer submatrices de una matriz mayor indicando la fila o columna que se desea. <samp>:</samp> indica todos los elementos.


In [56]:
A=[6 0 -3 8 9; 5 1 7 -4 2; 3 1 -6 2 -1] %Define matriz
A(2, :) %Fila 2 de la matriz a
A(:, 5) %Columna 5 de la matriz a
A(3, :)= -7 %Cambia todos los elementos de la fila 3 por -7

A =

   6   0  -3   8   9
   5   1   7  -4   2
   3   1  -6   2  -1

ans =

   5   1   7  -4   2

ans =

   9
   2
  -1

A =

   6   0  -3   8   9
   5   1   7  -4   2
  -7  -7  -7  -7  -7



<a name="H_A84CB336"></a>

***
#### Operaciones con matrices
***

Cuando trabajamos con matriz, para operar elemento a elemento hay que usar los comandos: `.*, ./, .^` (es decir, los operadores aritméticos con un punto delante), ya que el producto de matrices como tal no está definido elemento a elemento.


In [57]:
A=[-3 7; 2 5] %Define matriz
A+100 %Suma 100 a cada elemento de la matriz. % Igual para restas, multiplicaciones y divisiones
A=[-3 4 2; 1 7 -5] %Define matriz
B=[1 -2 -9; 4 2 6] %Define matriz
C=A+B % guarda el resultado en la variable c
A+B %Opera sin almacenar
0.3*A %Multiplica todos los valores de a por 0.3
C=A.*B %Multiplica matrices elemento a elemento

A =

  -3   7
   2   5

ans =

    97   107
   102   105

A =

  -3   4   2
   1   7  -5

B =

   1  -2  -9
   4   2   6

C =

  -2   2  -7
   5   9   1

ans =

  -2   2  -7
   5   9   1

ans =

  -0.9000   1.2000   0.6000
   0.3000   2.1000  -1.5000

C =

   -3   -8  -18
    4   14  -30



<a name="H_935550B0"></a>

***
#### Comandos sum y prod
***

-  El comando <samp>sum</samp>  se utiliza para calcular la suma de los elementos de un vector o matriz.
-  El comando <samp>prod</samp> se utiliza para calcular el producto de los elementos de un vector o matriz.

**Notas:**

-  Los comandos <samp>sum</samp> y <samp>prod</samp> se pueden aplicar a matrices y vectores.
-  Cuando se usa en una matriz sin especificar la dimensión,  los comandos operan por columnas.
-  Para sumar los elementos de cada fila, usa <samp>sum(M, 2)</samp>.
-  Para obtener el producto de los elementos de cada fila, usa <samp>prod(M, 2)</samp>.

In [58]:
M = [1, 2, 3;
     4, 5, 6;
     7, 8, 9];
suma_columnas = sum(M) % suma_columnas  [12, 15, 18]
suma_filas = sum(M, 2) % Calcula la suma de los elementos en cada fila

suma_columnas =

   12   15   18

suma_filas =

    6
   15
   24




**Ejercicio**: Calcula suma de todos los elementos de la matriz M.


In [59]:
% Escribe aquí el código:


<a name="H_703E17D1"></a>

### Ejercicios resueltos

a) Calcula la suma de los primeros 100 números naturales.


b) Calcula la suma de los cubos de los primeros 100 números naturales.


c) Calcula la suma de todos los números pares de 3 cifras y la suma de sus cuadrados.


d) Calcula el producto de las raíces cuadradas de los primeros 5 números naturales.


e) Calcula $\left(3^{\cos \;\frac{\pi }{4}} -2e^2 \right)\sqrt{\ln \;5}$


a) Calcula la suma de los primeros 100 números naturales.


In [60]:
k=1:100; % crea un vector con los números de 1 al 100.
suma=sum(k) % suma los elementos de k y guarda el resultado en la variable suma.

suma = 5050



b) Calcula la suma de los cubos de los primeros 100 números naturales.


In [61]:
k3=k.^3; % eleva al cubo cada elemento del vector k
suma=sum(k3) % suma los elementos de k3

suma = 2.5502e+07



c) Calcula la suma de todos los números pares de 3 cifras y la suma de sus cuadrados.


In [62]:
pares=100:2:998; % crea el vector de números pares de 3 cifras
suma=sum(pares) % suma los elementos del vector pares
suma_cuadrados=sum(pares.^2) % suma de los cuadrados de pares

suma = 247050
suma_cuadrados = 1.6601e+08



d) Calcula el producto de las raíces cuadradas de los primeros 5 números naturales.


In [63]:
a=1:5; % vector formado por los primeros 5 números naturales
raiz=sqrt(a); % calcula la raíz cuadrada de los elementos de a
producto=prod(raiz) % calcula la suma de las raíces cuadrada

producto = 10.954



e) Calcula $\left(3^{\cos \;\frac{\pi }{4}} -2e^2 \right)\sqrt{\ln \;5}$


In [64]:
% por partes:
a1 = 3^(cos(pi/4))
a2 = 2*exp(2)
a3 = sqrt(log(5))
res = (a1-a2)*a3

% directamente en una línea:
res = ( 3^(cos (pi/4) ) - 2*exp(2) )*sqrt( log(5) )

a1 = 2.1746
a2 = 14.778
a3 = 1.2686
res = -15.989
res = -15.989


<a name="H_A998F915"></a>

### Ejercicios propuestos
1.  Calcula la suma de las raíces cúbicas de los primeros 50 números naturales.
2. Calcula el producto de los cuadrados de todos los números impares de dos cifras.
3. Calcula:

$$\begin{array}{l} a=\left(\sqrt{6}-3\right)\left(3+2^{1/2} \cdot 3^{0.5} \right)\newline b=\;\cos (3\pi )+12\,sen\,\frac{3\pi }{2}\newline c=\;e^{4-8^{2/3} }  \end{array}$$

In [65]:
% Escribe aquí el código:





Compara tus resultados con las soluciones al final del notebook

<a name="H_0B4541F7"></a>

### Resumen

**Ventana de OCTAVE**


\- <samp>Command Window</samp>: Donde se escriben y ejecutan las órdenes.


\- <samp>Editor</samp>: Donde se escribe y guarda el código en scripts `.m`.


\- <samp>Workspace</samp>: Muestra las variables utilizadas.


**Variables y Tipos de Datos**


\- Crear variables: <samp>nombre = valor</samp>


\- Reglas para nombres de variables: Hasta 63 caracteres, comenzando con una letra, sin espacios, distinguibles entre mayúsculas y minúsculas.


**Comandos Básicos**

Se incluyen enlaces a la documentación online de los comandos en MatLab, que son equivalentes:

\-  [<samp>clear</samp>](https://es.mathworks.com/help/matlab/ref/clear.html) <samp>all/ clearvars</samp> : Limpia el espacio de trabajo.


\-[ <samp>clc</samp>](https://es.mathworks.com/help/matlab/ref/clc.html): Limpia la ventana de comandos.


\- [<samp>close</samp>](https://es.mathworks.com/help/matlab/ref/close.html) <samp>all</samp> : Cierra (todas) las figuras abiertas.


\- [<samp>whos</samp>](https://es.mathworks.com/help/matlab/ref/whos.html): Lista las variables definidas.


\- Operaciones básicas: Suma: `+`, Resta: `-`, Multiplicación: `*`, División: `/`, Potencia: `^`


\- [Funciones matemáticas elementales](https://es.mathworks.com/help/matlab/elementary-math.html):  <samp>exp(x), log(x), sin(x), cos(x), tan(x)</samp>, etc.


\- [<samp>format</samp>](https://es.mathworks.com/help/matlab/ref/format.html#btiwmh5-3) <samp>short</samp> : Muestra 4 decimales (long: Muestra 15 decimales; longe: Notación científica larga; rat: Muestra como fracción)


**Matrices y Vectores**


\- Crear vectores:


&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; \- Filas: <samp>u=[-7,3,4,-2]</samp> o <samp>v=[-7 3 4 -2]</samp>


&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; \- Columnas: <samp>w=[-7;3;4;-2]</samp>


&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; \- Secuencias: <samp>t=1:3:20</samp>, <samp>z=linspace(1,2,9)</samp>


\- Crear matrices: <samp>A=[3,-7;2,5]</samp>


\- [Operaciones con matrices](https://es.mathworks.com/help/matlab/matlab_prog/array-vs-matrix-operations.html?searchHighlight=suma%20resta&s_tid=srchtitle_support_results_2_suma%20resta) elemento a elemento: <samp>C=A+B</samp>, `C=A.\B`, `C=A.*B`


\- [<samp>length(v)</samp>](https://es.mathworks.com/help/matlab/ref/length.html) <samp>:</samp> Longitud del vector


\- [<samp>size(A)</samp>](https://es.mathworks.com/help/matlab/ref/size.html) <samp>:</samp> Tamaño de la matriz


\- [<samp>sum(A)</samp>](https://es.mathworks.com/help/matlab/ref/sum.html) <samp>:</samp> Calcular la suma de los elementos en un vector o matriz


\- [<samp>prod(A)</samp>](https://es.mathworks.com/help/matlab/ref/prod.html) <samp>:</samp> Producto de los elementos en un vector o matriz


\- [Matrices especiales](https://es.mathworks.com/help/matlab/matlab_oop/class-support-for-array-creation-functions.html):  <samp>zeros(3,5), ones(2,4), eye(4), diag([-7 2 4])</samp>


\- Extraer filas/columnas: <samp>A(2,:), A(:,5)</samp>


\- Modificar elementos: <samp>A(3,:)= -7</samp>


**Ayuda y Depuración**


\- Obtener ayuda: [<samp>help</samp>](https://es.mathworks.com/help/matlab/ref/help.html) <samp>function_name</samp>


\- Comentarios: Usar `%` para incluir comentarios en el código.


\- Ejecutar el código: "Run", "F9", o "Run Section".

<a name="H_AC6B0A09"></a>



### Soluciones

In [66]:
function [] = soluciones()

% Ejercicio 1:
k=1:50;
raices=k.^(1/3);
s1=sum(raices);
fprintf("Soluciones:\nEjercicio 1:\n S1 = %.4f\n",s1);

% Ejercicio 2:
k=11:2:99;
cuadrados = k.^2;
s2=prod(cuadrados);
fprintf("Ejercicio 2:\n S2 = %d\n",s2);

% Ejercicio 3
a = (sqrt(6) - 3)*(3 + 2^(1/2) * 3^(0.5));
b = cos(3*pi) + 12*sin(3*pi/2);
c = exp(4-8^(2/3));
fprintf("Ejercicio 3:\n a = %.0f\n b = %0.f\n c = %.0f\n",a,b,c);
end

Para ver las soluciones, ejecuta el comando

In [67]:
soluciones()

Soluciones:
Ejercicio 1:
 S1 = 139.7179
Ejercicio 2:
 S2 = 8.31753e+150
Ejercicio 3:
 a = -3
 b = -13
 c = 1
